### ISSUE TITLE
bug: X-Rq-Authorization was added to the request header for unknown reasons

### ISSUE THREAD

**Issue by mingxuandong** [requestly/requestly#1269]
### Describe your issue?

X-Rq-Authorization was added to the request header for unknown reasons, and it returned to normal after closing the requestly chrome extension.

### Repro steps

just enable requestly chrome extension

### What Requestly tool were you using?

- [X] Extension
- [ ] Desktop-App
- [ ] Android SDK
- [ ] Selenium

### Your Environment

windows 10 chrome 119

### Requestly Version

requestly chrome extension ver 23.12.10

### Error screenshot

![image](https://github.com/requestly/requestly/assets/37650077/8ab227ca-c187-40f0-82c5-9821d513e245)

**Comment by sachinjain024** [comment]
Thank you @mingxuandong for logging the issue. This looks like a regression bug happening as a fix for #1208 
We will try to fix this and release the next build asap.


@nafees87n Can we look at this on priority?

**Comment by nafees87n** [comment]
Hey @mingxuandong 
Can you please provide more information about this request. What was the response, status code etc.? Did this request fail because of CORS?

**Comment by HookyGV** [comment]
Can confirm. In my case the new header contains the value of the `Authorization` header. The CORS error detail:
```
A cross-origin resource sharing (CORS) request was blocked because it contained request headers that were neither CORS-safelisted (Accept, Accept-Language, Content-Language, Content-Type) nor allowed by the Access-Control-Allow-Headers response header of the associated preflight request.
To fix this issue, include the additional request headers you want to use in the Access-Control-Allow-Headers response header of the associated preflight request.
```
Attempting to remove the inserted header with Requestly also fails.

**Comment by sachinjain024** [comment]
@HookyGV Could you please help us reproduce this issue? 
- Could you share the website URL where can test this? 
- Do you have any Requestly Rule running trying to redirect this fetch request?

**Comment by HookyGV** [comment]
I had requestly installed, **and turned off**, and it was still adding this header. I've since uninstalled the extension and the CORS error goes away.

**Comment by sachinjain024** [comment]
@HookyGV It is a regression we are working on right away. Could you please help us with the URL where we can check it out?

**Comment by HookyGV** [comment]
Unfortunately no, it is happening on the secure side of our app. We use Requestly to allow `localhost` to pass through to our dev server for authentication.

**Comment by mingxuandong** [comment]
> Hey @mingxuandong Can you please provide more information about this request. What was the response, status code etc.? Did this request fail because of CORS?

yes, my request blocked by cors, beacuse the X-Rq-Authorization header added to request headers, and i have no rules enabled in requestly.
![image](https://github.com/requestly/requestly/assets/37650077/d1dad9c0-ae7a-4ab9-8b2f-0e9f9a48b3a6)

you can reproduce this behavior in any fetch request with an authorization request header.
just like this

![image](https://github.com/requestly/requestly/assets/37650077/3c562977-92e2-41e0-bdc5-599c5cddcdf3)

**Comment by mingxuandong** [comment]
> Thank you @mingxuandong for logging the issue. This looks like a regression bug happening as a fix for #1208 We will try to fix this and release the next build asap.
> 
> @nafees87n Can we look at this on priority?

For me, you can prioritize fixing any bug you want, I'm just here to report this issue. I currently have no work to do to modify network requests

**Comment by sachinjain024** [comment]
Thank you so much @mingxuandong! For now we are reverting the fix to #1208 to remove this regression

cc @nafees87n

**Comment by qkhanhpro** [comment]
> @HookyGV It is a regression we are working on right away. Could you please help us with the URL where we can check it out?

You can try to login to Oracle Cloud, for example. It breaks the whole Oracle Cloud access with tons of CORS even when I have no rule for Oracle

**Comment by ti83magic** [comment]
Can confirm as well, but what troubles me more is that this happens even when the extension is disabled.

![image](https://github.com/requestly/requestly/assets/1003563/c9eabbd0-7c48-4b11-8716-62efc44260bf)

**Comment by sachinjain024** [comment]
@ti83magic Yes, this is another regression that when Requestly is paused even then it adds the header. We will fix both issues in the next release. Please stay with us when we are rolling a new change out.

**Comment by ti83magic** [comment]
> @ti83magic Yes, this is another regression that when Requestly is paused even then it adds the header. We will fix both issues in the next release. Please stay with us when we are rolling a new change out.

Thanks for your quick reply and solution, I'll be sure to enable it again once it's been fixed.

**Comment by sachinjain024** [comment]
You can disable the extension for the time being by following these steps
1. Go to chrome://extensions
2. Disable Requestly

<img width="416" alt="Screenshot 2023-12-12 at 3 44 22 PM" src="https://github.com/requestly/requestly/assets/3108399/30c6c366-b0ca-4ee9-8c9b-6d064d70ab1e">

**Comment by nafees87n** [comment]
We are reverting the changes.
Meanwhile here's a workaround that can be used.
Use a Requestly Headers rule to add `X-Rq-Authorization` header to `Access-Control-Allow-Headers`.

Here's the sharedlist for the same: https://app.requestly.io/rules/shared-lists/viewer/1702378742503-Fix-x-rq-authorization-issue. You can import this rule directly from the sharedlist.
This should fix the issue temporarily.

<img width="775" alt="Screenshot 2023-12-12 at 5 03 12 PM" src="https://github.com/requestly/requestly/assets/56021937/771eac94-625d-4bdd-bab1-80e677aaf2b1">

**Comment by nafees87n** [comment]
The fix has been released with extension `v23.12.12`. It is available in the chrome store.

To manually update the extension you can follow these steps:
1. Go to URL: [chrome://extensions/](chrome://extensions/)
2. Turn on developer mode <img width="1440" alt="Screenshot 2023-12-13 at 12 07 09 PM" src="https://github.com/requestly/requestly/assets/56021937/da460775-b578-4e61-858c-8fa92625cbf1">
3. Click on the update button <img width="1439" alt="Screenshot 2023-12-13 at 12 07 38 PM" src="https://github.com/requestly/requestly/assets/56021937/0ecc0bd6-967d-441d-bf57-7c15f85b90a6">

Your Requestly extension should be updated to `v23.12.12` resolving this issue.

**Comment by mingxuandong** [comment]
This problem was solved after requestly chrome extension was upgraded to 23.12.12. I will close this issue now.
